data/data/bootstrap/baremetal/files/usr/local/bin/master-bmh-update.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+. /usr/local/bin/release-image.sh
+
+wait_for_existence() {
+    while [ ! -e "${1}" ]
+    do
+        sleep 5
+    done
+}
+
+# Wait for the API to become active
+until curl -s http://localhost:6385/v1/nodes; do
+    echo Waiting for API
+    sleep 20
+done
+
+# Wait for the nodes to become active after introspection.
+while curl -s http://localhost:6385/v1/nodes | jq '.nodes[] | .provision_state' | grep -v active; do
+    echo Waiting for nodes to become active
+    sleep 20
+done
+
+echo Nodes are all active
+echo Waiting for bootkube to finish
+
+# Wait for bootkube to come up.  We want to talk to the full instance not
+# the bootstrap openshift. This one only sleeps for 5 as we are getting
+# closer to being ready.
+wait_for_existence /opt/openshift/.bootkube.done
+
+echo Bootkube complete
+
+BAREMETAL_OPERATOR_IMAGE=$(image_for baremetal-operator)
+
+# Now we go through each image..
+for node in $(curl -s http://localhost:6385/v1/nodes | jq -r '.nodes[] | .uuid'); do
+    echo "host $node"
+    # And use the baremetal operator tool to load the introspection data into
+    # the BareMetalHost CRs as annotations, which BMO then picks up.
+    podman run --quiet --net=host \
+        --rm \
+        --name baremetal-operator \
+        --entrypoint /bin/get-hardware-details \
+        "${BAREMETAL_OPERATOR_IMAGE}" \
+        http://localhost:5050/v1 "$node"
```

Comment:
So I tried this out this morning, and this whole script looks pretty good. I think it'll work.  We don't have to use `kubectl patch`, we can directly call `annotate`.

Here's an example:

```bash
KUBECONFIG=/etc/kubernetes/kubeconfig

for node in $(curl -s http://localhost:6385/v1/nodes | jq -r '.nodes[] | .uuid'); do
    name=$(curl -H "X-OpenStack-Ironic-API-Version: 1.9" -s http://localhost:6385/v1/nodes/$node | jq -r .name)
    echo "Host $name, UUID: $node"
    # And use the baremetal operator tool to load the introspection data into
    # the BareMetalHost CRs as annotations, which BMO then picks up.
    HARDWARE_DETAILS=$(podman run --quiet --net=host \
        --rm \
        --name baremetal-operator \
        --entrypoint /get-hardware-details \
        "${BAREMETAL_OPERATOR_IMAGE}" \
        http://localhost:5050/v1 "$node" | jq '{hardware: .}')
        
     oc annotate -n openshift-machine-api baremetalhosts $name 'baremetalhost.metal3.io/status'='$HARDWARE_DETAILS'
done
```

Note, the entrypoint is /get-hardware-details, my mistake earlier saying /bin.

Reply:
Woot!  Works!  Getting annotations!